components/director/internal/model/runtime.go

```diff
@@ -103,13 +103,22 @@ type RuntimeInput struct {
 }
 
 func (i *RuntimeInput) ToRuntime(id string, tenant string) *Runtime {
+	if i == nil {
+		return nil
+	}
+
 	return &Runtime{
 		ID:          id,
 		Name:        i.Name,
 		Description: i.Description,
 		Tenant:      tenant,
 		Labels:      i.Labels,
 		Annotations: i.Annotations,
+		AgentAuth: &Auth{},
+		Status: &RuntimeStatus{
+			Condition: RuntimeStatusConditionInitial,
+			Timestamp: time.Now(),
```

Comment:
don't use `time.Now()` because it cannot be tested, introduce TImeProvider interface